apricot/local/serviceutil.go

```diff
@@ -84,7 +85,13 @@ func (s *Service) getStringMap(path string) map[string]string {
 			if v.Type() != cfgbackend.IT_Value {
 				continue
 			}
-			theMap[k] = v.Value()
+			if v.Value() == "\n" || k == "" || k == "\n" {
```

Comment:
We're getting there but not quite yet.

Firstly, here if the value is `"\n"`, you're just printing a warning but you aren't returning the value. The values should still be returned even if a warning is needed.

Secondly, you're also checking the key and not just the value, which was not required, and in fact is incorrect. We cannot just decide to return the value of a trimmed key or something like that (basically decreeing that a key with or without whitespace is the same), because then the assumptions behind what a *key* means are negated (e.g. uniqueness).

Thirdly, you are checking if the whole value is `"\n"`, which is not what we want, we discussed this. What you need is to check *whether the value has leading or trailing whitespace characters*, and if yes, warn and still use the value. One way to do this is with `string.HasPrefix/HasSuffix` calls, another, perhaps simpler, is to compare the string with the output of `strings.TrimSpace`. We also discussed returning the trimmed value or the verbatim value depending on a Viper setting to be added to Apricot as a final step, this is only possible if you indeed return some value after the check.